spec/latest/core/index.html

```diff
@@ -1250,8 +1250,48 @@ <h3>Description of HTTP Status Codes and Errors</h3>
       be given an identifier. When dereferenced, the <i>Error</i> resource can provide
       more detailed information or possible ways to resolve the problem, if applicable.<p/>
 
+    <p>Finally, server SHOULD provide error descriptions using an [[!RFC7807]] standard.
+      by using a <i>application/problem+json</i> response. When doing so server also MUST provide
+      an additional header pointing to either built-in Hydra <i>http://www.w3.org/ns/hydra/error</i>
+      context or any JSON-LD context that maps terms <i>type</i>, <i>title</i>, <i>detail</i>
+      and <i>instance</i> same way as the standard one.</p>
+
+    <pre class="example nohighlight" data-transform="updateExample"
+         title="RFC-7807 compatible error description">
+      <!--
+      HTTP/1.1 400 Bad Request
+      Content-Type: application/problem+json
+      Link: <http://www.w3.org/ns/hydra/error>; rel="http://www.w3.org/ns/json-ld#context"
+
+      {
+        "type": "https://example.com/probs/out-of-credit",
+        "@type": "http://www.w3.org/ns/hydra/core#Error",
+        "title": "You do not have enough credit.",
+        "detail": "Your current balance is 30, but that costs 50.",
+        "instance": "/account/12345/msgs/abc",
+        "balance": 30,
+        "accounts": ["/account/12345",
+                     "/account/67890"]
+      }
+      -->
+    </pre>
+
+    <p>While the built-in context makes the response fully compatible with the mentioned specification,
+      properties not defined in the standard Hydra's error context won't be visible for Hydra aware processor.
+      To overcome this, it is possible to declare an own context pointed the same way, that would combine standard
+      Hydra's error context and an additional JSON-LD context with either the <i>@vocab</i> or custom property mappings
+      telling the processor on how to interpret those custom error properties.</p>
+
+    <p>Resources provided may have an additional hint pointing to an <i>Error</i> type like in the example
+      above, but it is not mandatory to do so as all resources described with <i>application/problem+json</i>
+      are considered errors.</p>
+
+    <p>It is worth to mention that the <i>http://www.w3.org/ns/hydra/error</i> context does not map
+      the <i>status</i> property. It is due to possible discrepancies that may occur between actual status returned
+      and the status provided within the payload.</p>
```

Comment:
I don't quite understand the rationale for not mapping `status`?

Reply:
The status provided within the payload is not reliable - it may be different than the actual status from the response. I believe the payload should be supplemental for the protocol rather than cause possible discrepancies.

Reply:
That is irrelevant. The RFC requires that the property and actual status are the same. We should assume that they are 

> Generators MUST use the same status code in the actual HTTP response,
   to assure that generic HTTP software that does not understand this
   format still behaves correctly

Reply:
Agreed, @tpluscode.

Reply:
But this is part of the RFC-7807 spec, not hydra's spec. And I acknowledge this as a flaw - I'm aware that in some circumstances resulting response may still have a different value that the actual status due to i.e. proxy put in the middle that does the change (already saw this in action). I think not mapping the status directly to hydra term is a good move. Actual response should have the `status` property emitted to be compliant with the RFC-7807, but the resulting RDF resource could have that issue fixed.

I see that flaw somehow similar to requirement of having an URL dereferencable - there are circumstances that you cannot guarantee that.

Finally - pragmatic approach would be to see an actual status and not providing that value in the payload by default. This somehow discourages from looking it in the payload.

Reply:
> Actual response should have the status property emitted to be compliant with the RFC-7807, but the resulting RDF resource could have that issue fixed.
[...]
> Finally - pragmatic approach would be to see an actual status and not providing that value in the payload by default. This somehow discourages from looking it in the payload.

Rightly so; The RFC actually says that the `status` property is optional and only for convenience (or inconvenience?). From [section 3.1](https://tools.ietf.org/html/rfc7807#section-3.1)

> The "status" member, if present, is only advisory

Reply:
Having `status` in the payload makes it possible to transmit and store the payload otherwise than HTTP without losing any vital information. With the JSON-LD context we define for RFC 7807, we make `application/problem+json` responses available to RDF clients outside the realm of Hydra. I can envision a future where most RDF-enabled clients interpret `application/problem+json` according to Hydra's JSON-LD context. It is therefore vital that we provide a mapping of all properties defined in RFC 7807, regardless of them being optional or not.

Reply:
Totally. The RFC defines five standard fields. We must provide mapping for each one

Reply:
>>Having status in the payload makes it possible to transmit and store the payload otherwise than HTTP without losing any vital information.

Not really - the `type` is vital enough and carries all the details that client might need.

Anyway, I may agree that hydra could be the first one to address an issue of moving from `application/problem+json` to RDF world. We don't have too much options here:
- ignore `status` and leave it unmapped
- map `status` to `hydra:statusCode` and mark it as HTTP only and make it optional as it is in RFC-7807
- map `status` to other _low_level_ vocabulary that we could incorporate here; the only guess I have here would be to use `http:statusCodeNumber` from the HTTP Vocabulary in RDF 1.0, but I didn't see anyone using it; from RDF point of view it would also make the resource an `http:StatusCode` instance; I didn't find anything fitting - maybe `skos:notation` ?

Reply:
The `type` and every other field in problem+json are in fact optional.

There are 5 fields. We must map all of them. To keep `http` vocabulary as had been decided on other occasions, I vote for using `hydra:statusCode`

In the future, should a specific set of terms arise for Problem Details, we can make our term a subProperty as we did with member assertions

Reply:
Agreed, @tpluscode.

Reply:
I've reintroduced `hydra:statusCode` mapping for the `status` property and made few changes in the spec to note the possible issue with it. I hope it resolves the PR.